packages/core/src/codewhispererChat/tools/executeBash.ts

```diff
@@ -46,22 +156,67 @@ export class ExecuteBash {
         }
     }
 
-    public requiresAcceptance(): boolean {
+    public requiresAcceptance(): CommandValidation {
         try {
-            const args = ExecuteBash.parseCommand(this.command)
+            const args = split(this.command)
             if (!args || args.length === 0) {
-                return true
+                return { requiresAcceptance: true }
             }
 
-            if (args.some((arg) => dangerousPatterns.some((pattern) => arg.includes(pattern)))) {
-                return true
+            // Split commands by pipe and process each segment
+            let currentCmd: string[] = []
+            const allCommands: string[][] = []
+
+            for (const arg of args) {
+                if (arg === '|') {
+                    if (currentCmd.length > 0) {
+                        allCommands.push(currentCmd)
+                    }
+                    currentCmd = []
+                } else if (arg.includes('|')) {
+                    return { requiresAcceptance: true }
+                } else {
+                    currentCmd.push(arg)
+                }
+            }
+
+            if (currentCmd.length > 0) {
+                allCommands.push(currentCmd)
             }
 
-            const command = args[0]
-            return !readOnlyCommands.includes(command)
+            for (const cmdArgs of allCommands) {
+                if (cmdArgs.length === 0) {
+                    return { requiresAcceptance: true }
+                }
+
+                const command = cmdArgs[0]
+                const category = commandCategories.get(command)
+
+                switch (category) {
+                    case CommandCategory.Destructive:
+                        return { requiresAcceptance: true, warning: highRiskCommandWarningMessage }
```

Comment:
should it be `destructiveCommandWarningMessage` for this case? and high risk message for the one below?

Reply:
Fixed